src/lib/files/YarnFilesManager.ts

```diff
@@ -0,0 +1,97 @@
+import * as glob from 'globby'
+import { join, resolve } from 'path'
+import { PassThrough } from 'stream'
+import log from '../../logger'
+import { pathToFileObject } from './ProjectFilesManager'
+import { YarnSymlinkedModulesConfig } from './YarnLinkedFilesConfig'
+
+const stringToStream = (str: string) => {
+  const stream = new PassThrough()
+  stream.end(str)
+  return stream
+}
+
+export class YarnFilesManager {
+  private static LINKED_YARN_MODULES_IGNORED_FILES = [
+    '.DS_Store',
+    'README.md',
+    '.gitignore',
+    'CHANGELOG.md',
+    'node_modules/**',
+    '**/node_modules/**',
+  ]
+
+  public static async createFilesManager(projectSrc: string) {
+    const yarnLinkedModulesConfig = await YarnSymlinkedModulesConfig.createConfig(projectSrc)
+    return new YarnFilesManager(yarnLinkedModulesConfig)
+  }
+
+  private static async getFiles(npmModule: string, path: string) {
+    const files = await glob(['**'], {
+      cwd: path,
+      ignore: YarnFilesManager.LINKED_YARN_MODULES_IGNORED_FILES,
+      nodir: true,
+    })
+    return files.map(pathToFileObject(path, join('.linked_deps', npmModule))) as BatchStream[]
```

Comment:
I don't think it's clear why join with `.linked_deps`. Maybe adding a constant like
```
const BUILDER_HUB_LINKED_PATH = '.linked_deps'
```
can help